modal/_serialization.py

```diff
@@ -488,11 +526,21 @@ def deserialize_params(serialized_params: bytes, function_def: api_pb2.Function,
         # legacy serialization format - pickle of `(args, kwargs)` w/ support for modal object arguments
         param_args, param_kwargs = deserialize(serialized_params, _client)
     elif function_def.class_parameter_info.format == api_pb2.ClassParameterInfo.PARAM_SERIALIZATION_FORMAT_PROTO:
-        param_args = ()
-        param_kwargs = deserialize_proto_params(serialized_params, list(function_def.class_parameter_info.schema))
+        param_args = ()  # we use kwargs only for our implicit constructors
+        param_kwargs = deserialize_proto_params(serialized_params)
+        # TODO: We can probably remove the validation below since we do validation in the caller?
+        validate_params(param_kwargs, list(function_def.class_parameter_info.schema))
     else:
         raise ExecutionError(
             f"Unknown class parameter serialization format: {function_def.class_parameter_info.format}"
         )
 
     return param_args, param_kwargs
+
+
+def validate_parameter_type(parameter_type: type) -> "api_pb2.ParameterType.ValueType":
```

Comment:
Nit but I don't love it when "validate" functions have return values.

Reply:
Agreed - at first it didn't return anything and this was added in a lazy refactor commit - will adjust the name :)